### PR TITLE
Docker compose

### PR DESCRIPTION
**Pull request by dbuxton**
Placeholder for a full commit further down the road. Moving away from Vagrant as dev methodology to docker-compose

### CODE REVIEW

**Comment by dbuxton** [comment]
I think this is now nearly in a position to be merged.

The deployment strategy here is hopefully way simpler than we currently have. It will allow running on other Linux versions (and hopefully even Windows etc).

Instead of having very specific deployment strategies that rely on a specific version of Ubuntu (now outdated ;(, we can just do something like:

```
docker pull cabotapp/cabot:docker-compose
docker run -d --env-file YOUR_ENV_FILE cabotapp/cabot:docker-compose web
```

(If you want to try for yourself you can run the above, adding the tag `:docker-compose` to the image)

In `docker-entrypoint.sh` we have three process types specified (although you can do cleverer things if necessary): `web`, `worker` and `beatworker`. The minimum config will be one `web` container and one `beatworker` container (plus database and redis).

@JeanFred any thoughts on packaging improvements?

**Comment by JeanFred** [comment]
@dbuxton Thanks for your work on this! I’ll have a deeper look shortly, but overall it looks quite good.

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 from os import environ as env
 
+requirements_file = os.path.join(os.path.dirname(__file__), 'requirements.txt')
+with open(requirements_file) as f:
+    requirements = [line.rstrip('\n')
+                    for line in f
+                    if line and not line.startswith('#')]
+
 # pull in active plugins
 plugins = env['CABOT_PLUGINS_ENABLED'].split(',') if 'CABOT_PLUGINS_ENABLED' in env else ["cabot_alert_hipchat", "cabot_alert_twilio", "cabot_alert_email"]
```
Comment (JeanFred) [inline setup.py]: Shouldn’t this be read from `requirements-plugins.txt` as well?

**Comment by JeanFred** [comment]
I finished reviewing this − this all looks good!
I have added a `bin/build-app` shell script to ease things.

The commands needed to set-up would be:
```
docker-compose run --rm web bash bin/build-app
docker-compose run --rm web python manage.py createsuperuser
```

Packaging wise: I quite like the way Sentry does this with their ONBUILD hooks to allow for customisation of settings and plugins − I find it quite streamlined. https://github.com/getsentry/docker-sentry/blob/master/8.8/onbuild/Dockerfile 

If that’s fine with you, I’ll go ahead and merge this :)

File: production.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+web:
```
Comment (JeanFred) [inline production.yml]: Is this file still used?

**Comment by dbuxton** [comment]
Sounds good to me - we should update docs too...

**Comment by JeanFred** [comment]
Merged on the command line after rebasing and squashing commits. Thanks for all the work!

**Comment by dbuxton** [comment]
@JeanFred we should also update docs

**Pull request closed without merging**